src/main/java/com/salesforce/cdp/queryservice/util/QueryTokenExecutor.java

```diff
@@ -143,13 +143,15 @@ private int getMaxRetries(Properties properties) {
         }
     }
 
-    protected static OkHttpClient updateClientWithSocketFactory(OkHttpClient client, boolean isSocksProxyDisabled) {
+    protected OkHttpClient updateClientWithSocketFactory(OkHttpClient client, boolean isSocksProxyDisabled) {
+        OkHttpClient.Builder builder = client.newBuilder();
         if (isSocksProxyDisabled) {
-            return client.newBuilder()
-                    .socketFactory(new SFDefaultSocketFactoryWrapper(true))
-                    .build();
+            builder.socketFactory(new SFDefaultSocketFactoryWrapper(true));
         }
-        return client;
+        if(connection.addMetaDataInterceptor()) {
```

Comment:
Client once created will be reused for query as well as metadata call, how is this preventing that, i am confused there.

Reply:
But only executor created from metadata calls will have the interceptor attached.

Reply:
@KaviarasuSakthivadivel : The interceptor is reused for queries as well, which is not expected. One problem with that is, as the query urls are same, all the queries would result in same response with this approach, as the first response would get cached and the remaining responses would be coming from cache

Reply:
@vish689 I have modified the code as we discussed this earlier.